scripts/markdown/check-editlinks.js

```diff
@@ -0,0 +1,170 @@
+}
+
+/**
+ * Determines the appropriate edit link based on the URL path and file path
+ * @param {string} urlPath - The URL path to generate an edit link for
+ * @param {string} filePath - The actual file path
+ * @param {object[]} editOptions - Array of edit link options
+ * @returns {string|null} The generated edit link or null if no match
+ */
+function determineEditLink(urlPath, filePath, editOptions) {
+  // Remove leading 'docs/' if present for matching
+  const pathForMatching = urlPath.startsWith('docs/') ? urlPath.slice(5) : urlPath;
+
+  const target =
+    editOptions.find((edit) => pathForMatching.includes(edit.value)) || editOptions.find((edit) => edit.value === '');
+
+  if (!target) return null;
+
+  // Handle the empty value case (fallback)
+  if (target.value === '') {
+    return `${target.href}/docs/${urlPath}.md`;
+  }
+
+  // For other cases with specific targets
+  return `${target.href}/${path.basename(filePath)}`;
+}
+
+/**
+ * Recursively processes markdown files in a directory to generate paths and edit links
+ * @param {string} folderPath - The path to the folder to process
+ * @param {object[]} editOptions - Array of edit link options
+ * @param {string} [relativePath=''] - The relative path for URL generation
+ * @param {object[]} [result=[]] - Accumulator for results
+ * @returns {Promise<object[]>} Array of objects containing file paths and edit links
+ */
+async function generatePaths(folderPath, editOptions, relativePath = '', result = []) {
+  try {
+    const files = await fs.readdir(folderPath);
+
+    await Promise.all(
+      files.map(async (file) => {
+        const filePath = path.join(folderPath, file);
+        const relativeFilePath = path.join(relativePath, file);
+
+        // Skip _section.md files
+        if (file === '_section.md') {
+          return;
+        }
+
+        const stats = await fs.stat(filePath);
+
+        if (stats.isDirectory()) {
+          // Process directory
+          await generatePaths(filePath, editOptions, relativeFilePath, result);
+        } else if (stats.isFile() && file.endsWith('.md')) {
+          // Process all markdown files (including index.md)
+          const urlPath = relativeFilePath.split(path.sep).join('/').replace('.md', '');
+          result.push({
+            filePath,
+            urlPath,
+            editLink: determineEditLink(urlPath, filePath, editOptions)
+          });
+        }
+      })
+    );
+
+    return result;
+  } catch (err) {
+    console.error(`Error processing directory ${folderPath}:`, err);
+    throw err;
+  }
+}
+
+async function main() {
+  const editOptions = editUrls;
+
+  try {
+    const docsFolderPath = path.resolve(__dirname, '../../markdown/docs');
+    const paths = await generatePaths(docsFolderPath, editOptions);
+    console.log('Starting URL checks...');
+    const invalidUrls = await checkUrls(paths);
+
+    if (invalidUrls.length === 0) {
+      console.log('All URLs are valid.');
+      process.exit(0);
```

Comment:
We shouldn't use process.exit functions like this. Make the conditional rendering more appropriate. Like, make this if block for `invalidUrls` only.